tests/out/msl/binding-arrays.msl

```diff
@@ -146,11 +152,17 @@ fragment main_Output main_(
     metal::float4 _e244 = texture_array_bounded[non_uniform_index].sample(samp[non_uniform_index], uv, metal::level(0.0));
     v4_ = _e240 + _e244;
     metal::float4 _e248 = v4_;
-    texture_array_storage[0].write(_e248, metal::uint2(pix));
+    if (metal::all(metal::uint2(pix) < metal::uint2(texture_array_storage[0].get_width(), texture_array_storage[0].get_height()))) {
+        texture_array_storage[0].write(_e248, metal::uint2(pix));
+    }
     metal::float4 _e250 = v4_;
-    texture_array_storage[uniform_index].write(_e250, metal::uint2(pix));
+    if (metal::all(metal::uint2(pix) < metal::uint2(texture_array_storage[uniform_index].get_width(), texture_array_storage[uniform_index].get_height()))) {
+        texture_array_storage[uniform_index].write(_e250, metal::uint2(pix));
+    }
     metal::float4 _e252 = v4_;
-    texture_array_storage[non_uniform_index].write(_e252, metal::uint2(pix));
+    if (metal::all(metal::uint2(pix) < metal::uint2(texture_array_storage[non_uniform_index].get_width(), texture_array_storage[non_uniform_index].get_height()))) {
+        texture_array_storage[non_uniform_index].write(_e252, metal::uint2(pix));
+    }
```

Comment:
This doesn't seem to be checking the value of `non_uniform_index` at all - isn't that what `BoundsCheckPolicies::binding_array` is supposed to apply to?

Reply:
Correct. I haven't actually implemented bounds checking for the index into the array yet. This was merely to get binding indexing to play nice with existing bounds checks, and part of that was adding a new policy (that only works on unrestricted right now) so that the choose policy works correctly.